File: 1651-shuffle-string/solution.java
class Solution {
    public String restoreString(String s, int[] indices) {
        HashMap<Integer,Character> map= new HashMap<>();
        int n=indices.length;
        for(int i=0;i<n;i++){
            map.put(indices[i],s.charAt(i));
        }
        StringBuilder str= new StringBuilder("");
        for(int i=0;i<n;i++){
            str.append(map.get(i));
        }
        return str.toString();
    }
}

File: 1651-shuffle-string/solution.c
char* restoreString(char* s, int* indices, int indicesSize) {
    char *arr = malloc(sizeof(char)*(indicesSize+1));
    for(int i=0;i<indicesSize;i++){
        arr[indices[i]]=s[i];
    }
    arr[indicesSize]='\0';
    return arr;
}
